src/QsCompiler/DataStructures/SyntaxTree.fs

```diff
@@ -119,35 +119,70 @@ type QsLocation =
 
 
 /// used to represent the use of a type parameter within a fully resolved Q# type
+[<CustomEquality>]
+[<CustomComparison>]
 type QsTypeParameter =
     // TODO: origin needs adapting if we ever allow to declare type parameters on specializations
     {
-        /// the qualified name of the callable the type parameter belongs to
+        /// The qualified name of the callable the type parameter belongs to.
         Origin: QsQualifiedName
-        /// the name of the type parameter
+
+        /// The name of the type parameter.
         TypeName: string
-        /// the range at which the type parameter occurs relative to the statement (or partial statement) root
-        /// -> is Null for auto-generated type information, i.e. in particular for inferred type information
+
+        /// The source code range where the type parameter occurs. Null for auto-generated or inferred types. Range is
+        /// ignored when comparing type parameters.
+        // TODO: RELEASE 2021-10: Remove obsolete property.
+        [<Obsolete "Use ResolvedType.Range instead.">]
```

Comment:
I saw that in the NamespaceManager there are a lot of usages of this now deprecated handle. They are not showing up here because the file already contains a nowarn due to the access modifier changes. Please take a look at what updates are required there - I believe it should be possible to update it entire to use only the new handles.

Reply:
I'm not sure what you mean by a pre-existing nowarn for access modifiers? There is one in Namespace.fs but not NamespaceManager.fs. I added the nowarn to NamespaceManager.fs deliberately, because it relies on the range of UserDefinedType for attribute diagnostics. I think fixing that is a non-trivial change that I would like to do in a separate PR.

Reply:
Yes, my desire was to update the diagnostics generation. Making this a separate PR makes sense.